kafka-message-signing/src/main/kotlin/com/gxf/utilities/kafka/message/signing/MessageSigner.kt

```diff
@@ -226,6 +225,14 @@ class MessageSigner(properties: MessageSigningProperties) {
         }
     }
 
+    fun doAfterVerifyUsingHeader(consumerRecord: ConsumerRecord<String, out SpecificRecordBase>, action: Consumer<ConsumerRecord<String, out SpecificRecordBase>>) {
```

Comment:
What is the purpose of this method? (Should it be the responsibility of this class to perform an action after verification?)
Should a similar method be implemented for the `verifyUsingField` method?

Reply:
The purpose was to reduce the amount of code duplication in using verifyUsingHeader with two different avro schemas. However, on closer inspection, the Consumer does not reduce the amount of code needed. I will rewrite this as a verify method that throws a VerificationException instead of returning a boolean when the signature is wrong.